geode-core/src/test/java/org/apache/geode/internal/cache/ReplicateEntryIdleExpirationTest.java

```diff
@@ -0,0 +1,127 @@
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.geode.cache.ExpirationAction.DESTROY;
+import static org.apache.geode.cache.RegionShortcut.REPLICATE;
+import static org.apache.geode.test.dunit.Host.getHost;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.google.common.base.Stopwatch;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.ExpirationAttributes;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.test.dunit.AsyncInvocation;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.CacheRule;
+import org.apache.geode.test.dunit.rules.DistributedTestRule;
+import org.apache.geode.test.junit.categories.DistributedTest;
+
+@Category(DistributedTest.class)
+@SuppressWarnings("serial")
+public class ReplicateEntryIdleExpirationTest implements Serializable {
+
+  @ClassRule
+  public static DistributedTestRule distributedTestRule = new DistributedTestRule();
+
+  private static final AtomicBoolean KEEP_READING = new AtomicBoolean(true);
+
+  protected static final String KEY = "KEY";
+  protected static final String VALUE = "VALUE";
+
+  protected final VM member1 = getHost(0).getVM(0);
+  protected final VM member2 = getHost(0).getVM(1);
+  protected final VM member3 = getHost(0).getVM(2);
+  protected final String regionName = getClass().getSimpleName();
+
+  @Rule
+  public CacheRule cacheRule = CacheRule.builder().createCacheIn(member1).createCacheIn(member2)
+      .createCacheIn(member3).build();
+
+  @Before
+  public void setUp() throws Exception {
+    VM[] vms = new VM[] {member1, member2, member3};
+    for (VM vm : vms) {
+      vm.invoke(() -> {
+        ExpiryTask.suspendExpiration();
+        createRegion();
+      });
+    }
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    VM[] vms = new VM[] {member1, member2, member3};
+    for (VM vm : vms) {
+      vm.invoke(() -> {
+        KEEP_READING.set(false);
```

Comment:
This should also be in setUp(), right? Now the first test method sees the KEEP_READING value to be true and other subsequent tests (if added in future) sees it set to FALSE...

Reply:
KEEP_READING is now initialized to true in setUp